src/Components/Members/Edit/EditMembers.jsx

```diff
@@ -0,0 +1,253 @@
+import Modal from '../../Shared/Modal';
+
+const MembersEdit = () => {
+  const [member, setMember] = useState([]);
+  const [message, setMessage] = useState('');
+  const [showModal, setShowModal] = useState(false);
+
+  const { id } = useParams();
+
+  const getMember = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${id}`);
+      const data = await response.json();
+      setMember(data.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    getMember(id);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleFirstName = (e) => {
+    setMember({ ...member, firstName: e.target.value });
+  };
+
+  const handleLastName = (e) => {
+    setMember({ ...member, lastName: e.target.value });
+  };
+
+  const handleDniChange = (e) => {
+    setMember({ ...member, dni: e.target.value });
+  };
+
+  const handlePhoneChange = (e) => {
+    setMember({ ...member, phone: e.target.value });
+  };
+
+  const handleEmailChange = (e) => {
+    setMember({ ...member, email: e.target.value });
+  };
+
+  const handleCityChange = (e) => {
+    setMember({ ...member, city: e.target.value });
+  };
+
+  const handleBirthdayChange = (e) => {
+    setMember({ ...member, birthDay: e.target.value });
+  };
+
+  const handlePostalCodeChange = (e) => {
+    setMember({ ...member, postalCode: e.target.value });
+  };
+
+  const handleMembershipChange = (e) => {
+    setMember({ ...member, membership: e.target.value });
+  };
+
+  const handleIsActiveChange = (e) => {
+    setMember({ ...member, isActive: e.target.value });
+  };
+
+  const updateMember = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          firstName: member.firstName,
+          lastName: member.lastName,
+          dni: member.dni,
+          phone: member.phone,
+          email: member.email,
+          city: member.city,
+          birthDay: member.birthDay,
+          postalCode: member.postalCode,
+          isActive: member.isActive,
+          membership: member.membership
+        })
+      });
+
+      const data = await response.json();
+      setMessage(data.message);
+      console.log(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    updateMember(id);
+    setShowModal(true);
+  };
+
+  return (
+    <div className={styles.container}>
+      <Modal
+        onClose={() => setShowModal(false)}
+        isOpen={showModal}
+        title={message}
+        success={true}
+      />
+      ;
+      <div>
+        <h3 className={styles.whiteLetters}>Edit current member</h3>
+        <form onSubmit={(e) => handleSubmit(e)}>
+          <div>
+            {member ? (
```

Comment:
It will never happen that 'member' is not defined because you initially defined it as an empty array. If you want the 'loading' to appear while the data is not loaded, you should use (member.length === 0).

Reply:
Done. Thanks you, I didn't notice that at all